pkg/serverless/flush/strategy.go

```diff
@@ -0,0 +1,95 @@
+package flush
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+)
+
+// Strategy is deciding whether the data should be flushed or not at the given moment.
+type Strategy interface {
+	String() string
+	ShouldFlush(moment Moment, t time.Time) bool
+}
+
+// Moment represents at which moment we're asking the flush strategy if we
+// should flush or not.
+// Note that there is no entry for the shutdown of the environment because we always
+// flush in this situation.
+type Moment string
```

Comment:
Why a string and not an int ?

Reply:
If used in a debug log or anything, it's way more readable (even while  `%+v` for the developer), and since there like 1 or 2 instances of this at runtime, we don't mind about memory usage.